src/allocator.f90

```diff
@@ -33,17 +35,14 @@ module m_allocator
      !! [[m_allocator(module):release_block(subroutine)]].  The
      !! released block is then pushed in front of the block list.
 
-    integer :: ngrid, sz
+    integer :: ngrid
     !> The id for the next allocated block.  This counter is
     !> incremented each time a new block is allocated.
     integer :: next_id = 0
-    !> Padded dimensions for x, y, and z oriented fields
-    integer :: xdims_padded(3), ydims_padded(3), zdims_padded(3)
-    !> Padded dimensions for natural Cartesian ordering
-    integer :: cdims_padded(3)
     !> The pointer to the first block on the list.  Non associated if
     !> the list is empty
     ! TODO: Rename first to head
+    type(mesh_t) :: mesh
```

Comment:
`mesh` is a member in allocator, however only the `get_padded_dims` method is being used inside. Maybe we can still pass the local domain size to the allocator and let it sort out the padding, and capture this info and store a copy in `mesh`. This way we won't need to have a copy of the mesh object in the allocator.
One use case of this in the future: If we have two separate backends (CUDA+OpenMP) working at the same time, we may want to allow them to work with their preferred SZ. Then `mesh%sz` won't be as global as we currently think it is. So it might be better to pass a `sz` when instantiating the allocator object and have it sort out the padding, and teach the padding to the `mesh` object instead.